### PR TITLE
Dual finance staking option into governance UI

### PR DESCRIPTION
**Pull request by brittcyr**
Add an instruction for doing a staking option

### CODE REVIEW

**Comment by abrzezinski94** [comment]
It would be good to also add some decoding to have nice vaules in proposals example Projects/governance-ui/components/instructions/programs/voteStakeRegistry.tsx

File: pages/dao/[symbol]/proposal/components/instructions/Dual/StakingOption.tsx [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+          onChange={(evt) =>
+            handleSetForm({
+              value: evt.target.value,
+              propertyName: 'soName',
+            })
+          }
+          error={formErrors['soName']}
+        />
+      </Tooltip>
+      <Tooltip content="Treasury owned account providing the assets for the option. When the recipient exercises, these are the tokens they receive.">
```
Comment (DonDuala) [inline pages/dao/[symbol]/proposal/components/instructions/Dual/StakingOption.tsx:87]: Add to tooltip
"For SOL/USDC Calls, enter SOL. For SOL/USDC Puts, enter USDC."

File: pages/dao/[symbol]/proposal/components/instructions/Dual/StakingOption.tsx [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+          onChange={(value) => {
+            handleSetForm({ value, propertyName: 'baseTreasury' })
+          }}
+          value={form.baseTreasury}
+          error={formErrors['baseTreasury']}
+          shouldBeGoverned={shouldBeGoverned}
+          governance={governance}
+        ></GovernedAccountSelect>
+      </Tooltip>
+      <Tooltip content="Treasury owned account receiving payment for the option exercise. This is where payments from exercise accumulate.">
```
Comment (DonDuala) [inline pages/dao/[symbol]/proposal/components/instructions/Dual/StakingOption.tsx:100]: Add to tooltip
"For SOL/USDC Calls, enter USDC. For SOL/USDC Puts, enter SOL."

File: pages/dao/[symbol]/proposal/components/instructions/Dual/StakingOption.tsx [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+          onChange={(evt) =>
+            handleSetForm({
+              value: evt.target.value,
+              propertyName: 'optionExpirationUnixSeconds',
+            })
+          }
+          error={formErrors['optionExpirationUnixSeconds']}
+        />
+      </Tooltip>
+      <Tooltip content="Strike price for the staking option. Units are atomic quote atoms per lot">
```
Comment (DonDuala) [inline pages/dao/[symbol]/proposal/components/instructions/Dual/StakingOption.tsx:141]: Clarify what to do for puts by inverting the strike in USDC terms

**Review by DonDuala — COMMENTED** [review]
Clarify call vs put entries

File: pages/dao/[symbol]/proposal/components/instructions/Dual/StakingOption.tsx (hunk 3 above)
Comment (brittcyr) [reply pages/dao/[symbol]/proposal/components/instructions/Dual/StakingOption.tsx]: Already says that, but added an example to be more clear

**Review by DonDuala — APPROVED** [review]

**Comment by DonDuala** [comment]
Logo-cropped

File: hooks/useGovernanceAssets.ts [hunk 4]
```diff
@@ -169,6 +169,10 @@ export default function useGovernanceAssets() {
     [PackageEnum.Common]: {
       name: 'Common',
     },
+    [PackageEnum.Dual]: {
+      name: 'Dual',
+      image: '/img/dual-logo.png',
+    },
```
Comment (GregoryNEUT) [inline hooks/useGovernanceAssets.ts:175]: Should we use full name?

```suggestion
    [PackageEnum.DualFinance]: {
      name: 'Dual Finance',
      image: '/img/dual-logo.png',
    },
```

Before:
Screenshot 2022-11-30 at 17 30 11

After:
Screenshot 2022-11-30 at 17 30 53

File: hooks/useGovernanceAssets.ts [hunk 5]
```diff
@@ -366,6 +370,20 @@ export default function useGovernanceAssets() {
       packageId: PackageEnum.Common,
     },
 
+    /*
+      ██████  ██    ██  █████   ██      
+      ██   ██ ██    ██ ██   ██  ██      
+      ██   ██ ██    ██ ███████  ██      
+      ██   ██ ██    ██ ██   ██  ██      
+      ██████   ██████  ██   ██  ███████ 
+    */
```
Comment (GregoryNEUT) [inline hooks/useGovernanceAssets.ts:379]:
```suggestion
    /*
      ██████  ██    ██  █████  ██          ███████ ██ ███    ██  █████  ███    ██  ██████ ███████ 
      ██   ██ ██    ██ ██   ██ ██          ██      ██ ████   ██ ██   ██ ████   ██ ██      ██      
      ██   ██ ██    ██ ███████ ██          █████   ██ ██ ██  ██ ███████ ██ ██  ██ ██      █████   
      ██   ██ ██    ██ ██   ██ ██          ██      ██ ██  ██ ██ ██   ██ ██  ██ ██ ██      ██      
      ██████   ██████  ██   ██ ███████     ██      ██ ██   ████ ██   ██ ██   ████  ██████ ███████                                                                                        
    */
```

File: package.json [hunk 6]
```diff
@@ -38,6 +38,7 @@
     "@civic/solana-gateway-react": "0.11.0",
     "@dialectlabs/react-sdk-blockchain-solana": "1.0.0-beta.3",
     "@dialectlabs/react-ui": "1.1.0-beta.5",
+    "@dual-finance/staking-options": "^0.0.3",
```
Comment (GregoryNEUT) [inline package.json:41]: Would it be possible to target a specific version of the sdk?

Latest is `0.0.7` 

https://www.npmjs.com/package/@dual-finance/staking-options

File: pages/dao/[symbol]/proposal/components/instructions/Dual/StakingOption.tsx [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+import { NewProposalContext } from '../../../new'
+import GovernedAccountSelect from '../../GovernedAccountSelect'
+import { Governance } from '@solana/spl-governance'
+import useGovernanceAssets from '@hooks/useGovernanceAssets'
+import { ProgramAccount } from '@solana/spl-governance'
+import Input from '@components/inputs/Input'
+import getConfigInstruction from '@utils/instructions/Dual'
+import useWalletStore from 'stores/useWalletStore'
+import { getDualFinanceStakingOptionSchema } from '@utils/validations'
+import Tooltip from '@components/Tooltip'
```
Comment (GregoryNEUT) [inline pages/dao/[symbol]/proposal/components/instructions/Dual/StakingOption.tsx:16]: nit, deduplicate `@solana/spl-governance` import: 

```suggestion
import React, { useContext, useEffect, useState } from 'react'
import { ProgramAccount, Governance } from '@solana/spl-governance'
import {
  UiInstruction,
  DualFinanceStakingOptionForm,
} from '@utils/uiTypes/proposalCreationTypes'
import { NewProposalContext } from '../../../new'
import GovernedAccountSelect from '../../GovernedAccountSelect'
import useGovernanceAssets from '@hooks/useGovernanceAssets'
import Input from '@components/inputs/Input'
import getConfigInstruction from '@utils/instructions/Dual'
import useWalletStore from 'stores/useWalletStore'
import { getDualFinanceStakingOptionSchema } from '@utils/validations'
import Tooltip from '@components/Tooltip'
```

File: pages/dao/[symbol]/proposal/components/instructions/Dual/StakingOption.tsx [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+  async function getInstruction(): Promise<UiInstruction> {
+    const obj: UiInstruction = await getConfigInstruction({
+      connection,
+      form,
+      schema,
+      setFormErrors,
+      wallet,
+    })
+    return obj
+  }
```
Comment (GregoryNEUT) [inline pages/dao/[symbol]/proposal/components/instructions/Dual/StakingOption.tsx:59]: nit: remove async wrapping

```suggestion
  function getInstruction(): Promise<UiInstruction> {
    return getConfigInstruction({
      connection,
      form,
      schema,
      setFormErrors,
      wallet,
    })
  }
```

File: utils/instructions/Dual/index.ts [hunk 9]
```diff
@@ -0,0 +1,219 @@
...
+      serializedInstruction,
+      isValid: true,
+      prerequisiteInstructions: prerequisiteInstructions,
+      prerequisiteInstructionsSigners: [helperTokenAccount],
+      governance: form.baseTreasury?.governance,
+      additionalSerializedInstructions,
+      chunkSplitByDefault: true,
+      chunkBy: 2,
+    }
+    return obj
```
Comment (GregoryNEUT) [inline utils/instructions/Dual/index.ts:207]: Nit, no need to use an intermediary variable to type the object. The returned value of the function `getConfigInstruction` is typed:

```suggestion
    return  {
      serializedInstruction,
      isValid: true,
      prerequisiteInstructions: prerequisiteInstructions,
      prerequisiteInstructionsSigners: [helperTokenAccount],
      governance: form.baseTreasury?.governance,
      additionalSerializedInstructions,
      chunkSplitByDefault: true,
      chunkBy: 2,
    }
```

**Pull request merged**